src/vunnel/provider.py

```diff
@@ -192,8 +194,14 @@ def _fetch_or_use_results_archive(self) -> tuple[list[str], int, datetime.dateti
         if not latest_entry:
             raise RuntimeError("no listing entry found")
 
-        if self._has_newer_archive(latest_entry=latest_entry):
+        if self._has_newer_archive(latest_entry=latest_entry) or self.runtime_cfg.skip_newer_archive_check:
```

Comment:
```suggestion
        if self.runtime_cfg.skip_newer_archive_check or self._has_newer_archive(latest_entry=latest_entry):
```
I think flipping the order has the added benefit of not calling `self._has_newer_archive()` if the config flag is already trie.